week3/Borislav/issue_TAPC-3.py

```diff
@@ -0,0 +1,46 @@
+# -*-config: utf_8 -*-
+
+
+# The function of which we read docstring
+def som_func():
+    """
+    Calculate the square root of a number.
+
+    Accepts:
+    n: the number to get the square root of.
+    Returns:
+    the square root of n.
+    Raises:
+    TypeError: if n is not a number.
+    ValueError: if n is negative.
+    """
+    pass
+
+
+def doc_validation(func):
+    """"
+    Function decorator
+    Records the results of the validation of the docstring in the log file
+    log.txt
+    """
+    def wrapper_function(*args, **kwargs):
+
+        doc_str = str(args)
```

Comment:
Please re-factor the decorator so it reads the docstring directly from the decorated function and not by reading as argument for a function that reads the docstring of another function. The added complexity is unnecessary.